packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -539,7 +539,7 @@ export class SchemaFactory<
 	/**
 	 * Make a field of type identifier instead of the default which is required.
 	 */
-	public get identifier(): FieldSchema<FieldKind.Identifier> {
+	public get identifier(): FieldSchema<FieldKind.Identifier, typeof this.string> {
```

Comment:
@daesunp - this line here is what changes the output type from `unknown` to `string | undefined`. The rest of the changes in this PR, to `ApplyKind`, are what make it not-undefined when reading (so, just `string`, rather than `string | undefined`).